### PR TITLE
Allow specifying the config file on the command line

### PR DESCRIPTION
**Pull request by jamesfoster**
This allows you to run the simulator multiple times with different configurations. Enabling you to store your config in source control, for example, without having to also store the exe in the same folder.

### CODE REVIEW

File: src/AzureEventGridSimulator/Program.cs [hunk 1]
```diff
@@ -14,46 +14,53 @@ namespace AzureEventGridSimulator
 {
     public static class Program
     {
-        public static void Main()
+        public static void Main(string[] args)
         {
             try
             {
-                var host = WebHost.CreateDefaultBuilder()
```
Comment (jamesfoster) [inline src/AzureEventGridSimulator/Program.cs]: Avoiding long lines by removing the hanging indentation

File: src/AzureEventGridSimulator/Program.cs [hunk 2]
```diff
@@ -14,46 +14,53 @@ namespace AzureEventGridSimulator
...
+                    .CreateDefaultBuilder(args)
+                    .ConfigureAppConfiguration((context, builder) =>
+                    {
+                        var env = context.HostingEnvironment;
 
-                                      var config = builder.AddJsonFile("appsettings.json", false, false)
-                                                          .AddJsonFile($"appsettings.{env.EnvironmentName.ToLowerInvariant().Trim()}.json", true, false)
-                                                          .AddEnvironmentVariables()
-                                                          .Build();
+                        var config = builder.Build();
```
Comment (jamesfoster) [inline src/AzureEventGridSimulator/Program.cs]: By calling `CreateDefaultBuilder`, the configuration already reads from these 3 sources. No need to add them again. The only difference now is that `appsettings.json` is no longer required (by default it is added as optional)

File: src/AzureEventGridSimulator/Program.cs [hunk 3]
```diff
@@ -14,46 +14,53 @@ namespace AzureEventGridSimulator
...
-                                  })
-                                  .UseStartup<Startup>()
-                                  .UseSerilog()
-                                  .UseKestrel(options =>
-                                  {
-                                      var simulatorSettings = (SimulatorSettings)options.ApplicationServices.GetService(typeof(SimulatorSettings));
+                        var configFile = config.GetValue<string>("ConfigFile");
+                        if (!string.IsNullOrEmpty(configFile))
+                        {
+                            builder.AddJsonFile(configFile, optional: false);
```
Comment (jamesfoster) [inline src/AzureEventGridSimulator/Program.cs]: If you specify the `ConfigFile` setting we add that as a config source as not optional.

We *could* say that if you don't specify that option we add `appsettings.json` as not optional?

Comment (pmcilreavy) [reply src/AzureEventGridSimulator/Program.cs]: Yeah that sounds reasonable.

Comment (pmcilreavy) [reply src/AzureEventGridSimulator/Program.cs]: I've not even tried running it without a appsettings.json file present.

**Comment by pmcilreavy** [comment]
I'm happy to merge this one if you've no other changes to push.

**Comment by jamesfoster** [comment]
Apols. Been on holiday. Thanks for this, I'm glad it passed your review! :)

**Pull request merged**